### PR TITLE
obs-outputs: Fix provision of CA certificates to mbedTLS on macOS

### PR DESCRIPTION
**Pull request by PatTheMav**
### Description
Changes macOS-specific code for provision of CA certificates to mbedtls to use `SecTrustCopyAnchorCertificates`.

### Motivation and Context
Fixes https://github.com/obsproject/obs-studio/issues/9315

### How Has This Been Tested?
Tested locally by streaming to YouTube and checking contents of certificate references in debugger.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by derrod — APPROVED** [review]
Can confirm this fixed the issue on my system.

File: plugins/obs-outputs/librtmp/rtmp.c [hunk 1]
```diff
@@ -315,36 +315,29 @@ RTMP_TLS_LoadCerts(RTMP *r) {
...
 
         if (data && length > 0) {
             mbedtls_x509_crt_parse_der(chain, data, length);
         }
+        
+        CFRelease(der_data);
     }
-
-    CFRelease(result);
+
```
Comment (gxalpha) [inline plugins/obs-outputs/librtmp/rtmp.c:339]: Probably unintentional whitespaces

**Review by gxalpha — APPROVED** [review]
Fixes the issue here as well.

File: plugins/obs-outputs/librtmp/rtmp.c [hunk 2]
```diff
@@ -315,36 +315,29 @@ RTMP_TLS_LoadCerts(RTMP *r) {
...
-        size_t length = CFDataGetLength(data_ref);
+        CFDataRef der_data = SecCertificateCopyData(cert);
+
+        const UInt8 *data = CFDataGetBytePtr(der_data);
+        size_t length = CFDataGetLength(der_data);
 
         if (data && length > 0) {
             mbedtls_x509_crt_parse_der(chain, data, length);
         }
+
```
Comment (gxalpha) [inline plugins/obs-outputs/librtmp/rtmp.c:336]: There's a trailing tab here as well

**Review by RytoEX — COMMENTED** [review]
Commit message nit:
* mbedtls -> mbedTLS

Technically, I think they went with "Mbed TLS" some time ago, but we currently still use "mbedTLS" in descriptive text.

**Comment by yuriy-chumak** [comment]
Confirm. It works now.

**Pull request merged**